File: app/src/main/java/com/example/aiko/data/remote/ApiService.kt
package com.example.aiko.data.remote

import com.example.aiko.data.model.Position
import com.example.aiko.data.model.StopBusItem
import retrofit2.http.GET
import retrofit2.http.POST

interface ApiService {
    @GET("Posicao")
    suspend fun getPosition(): Position

    @GET("Parada/Buscar?termosBusca=")
    suspend fun getStopBus(): List<StopBusItem>

    @POST("Login/Autenticar?token=402b9ee6bea7d98f703c99da5b4be40ddf458e016b23cd83598247223356083c")
    suspend fun postAuth(): Boolean
}

File: teste-aiko/src/services/api-service.ts
/* eslint-disable import/order */
/* eslint-disable prettier/prettier */
import axios from 'axios';
import { OLHO_VIVO_BASE_URL, OLHO_VIVO_API_KEY } from '@env';
import { Linha, Parada, Posicao, PrevisaoParadaResponse } from '../types/types';

const api = axios.create({
  baseURL: OLHO_VIVO_BASE_URL,
});

export const login = async (): Promise<void> => {
  const response = await api.post('/Login/Autenticar', null, {
    params: { token: OLHO_VIVO_API_KEY },
  });
  if (!response.data) {
    throw new Error('Autenticação falhou');
  }
};

export const getLinhas = async (): Promise<Linha[]> => {
  await login();

  const response = await api.get('/Linha/Buscar');
  return response.data;
};
export const getPosicoes = async (): Promise<Linha[]> => {
  await login();

  const response = await api.get('/Posicao');
  return response.data.l;
};
export const getParadasByLine = async (linha: number): Promise<Parada[]> => {
  await login();

  const response = await api.get(`/BuscarParadasPorLinha?codigoLinha=${linha}`);
  return response.data;
};
export const getAllParadas = async (): Promise<Parada[]> => {
  await login();

  const response = await api.get(`/Parada/Buscar?termosBusca=`);
  return response.data;
};
export const getPrevisaoChegada = async (
  paradaId: number
): Promise<PrevisaoParadaResponse> => {
  await login();

  const response = await api.get(`/Previsao/Parada?codigoParada=${paradaId}`);
  return response.data;
};
